grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/ArmeriaChannel.java

```diff
@@ -104,10 +105,18 @@
         final int maxOutboundMessageSizeBytes = options.get(GrpcClientOptions.MAX_OUTBOUND_MESSAGE_SIZE_BYTES);
         final int maxInboundMessageSizeBytes = maxInboundMessageSizeBytes(options);
         final boolean unsafeWrapResponseBuffers = options.get(GrpcClientOptions.UNSAFE_WRAP_RESPONSE_BUFFERS);
+        final Compressor compressor = options.get(GrpcClientOptions.COMPRESSOR);
+        final DecompressorRegistry decompressorRegistry = options.get(GrpcClientOptions.DECOMPRESSOR_REGISTRY);
 
         final HttpClient client;
 
-        final CallCredentials credentials = callOptions.getCredentials();
+        CallCredentials credentials = callOptions.getCredentials();
+        if (credentials == null) {
```

Comment:
Q) is there no need to also check equality with `NullCallCredentials.INSTANCE` here?

```suggestion
        if (credentials == null || credentials == NullCallCredentials.INSTANCE) {
```